lib/get-ask-message.js

```diff
@@ -0,0 +1,132 @@
+const fs = require('fs')
+const path = require('path')
+const { promisify } = require('util')
+const { spawn } = require('child_process')
+const inquirer = require('inquirer')
+const dedent = require('dedent')
+const tmp = require('tmp')
+
+module.exports = getAskMessage
+
+const access = promisify(fs.access)
+const writeFile = promisify(fs.writeFile)
+const readFile = promisify(fs.readFile)
+const stat = promisify(fs.stat)
+const unlink = promisify(fs.unlink)
+
+function getDefaultEditor () {
+  const platformDefault = process.platform === 'win32' ? 'notepad' : 'vim'
+  return process.env.EDITOR || process.env.VISUAL || platformDefault
+}
+
+async function runEditor (editor, filename) {
+  const args = []
+  if (editor === 'vim') {
+    args.push('+star') // start vim in insert mode
+    args.push('+') // and on the last line
+  }
+  args.push(filename)
+
+  const startStat = await stat(filename)
+
+  const proc = spawn(editor, args, { stdio: 'inherit' })
+  const startTime = Date.now()
+  await new Promise((resolve, reject) => {
+    proc.on('error', reject)
+    proc.on('exit', () => resolve())
+  })
+
+  const endStat = await stat(filename)
+  const hasSaved = endStat.mtime > startStat.mtime
+
+  // If it exited really quickly without updating the file, it might have forked, and we should wait for the message to be written
+  // (visual editors like VS Code like to do this)
+  if (Date.now() - startTime < 2000 && !hasSaved) {
+    const prompt = await inquirer.prompt({
+      type: 'confirm',
+      name: 'continue',
+      message: 'Could not determine whether the editor is closed. Please type your message and save the file, and then press Y or Enter to continue. Press N to abort.',
+      default: true
+    })
+
+    if (!prompt.continue) {
+      throw new Error('Empty message--aborting Ask.')
+    }
+  }
+}
+
+async function getAskMessage (dirname) {
+  const basename = path.basename(dirname)
+  const editor = getDefaultEditor()
+  let header = dedent`
+    # Asking for help with: ${basename}
+    # Please enter your question below. There is no specific formatting, but
+    # feel free to use Markdown if you need to.
+  `
+  if (editor === 'vim') {
+    header += '\n' + dedent`
+      #
+      # You've been dropped into vim, already in 'insert mode'.
+      # After writing your message, press Escape, and then type ':wq' (no quotes) to save and exit.
```

Comment:
```suggestion
      # After writing your message, to save and exit, press Escape then type :wq
```
Shorter and means we don't need quotes and "no quotes"

Reply:
clever :)